### PR TITLE
Add new route to get team users and groups

### PR DESCRIPTION
**Pull request by wpjunior**

### CODE REVIEW

**Review by nettoclaudio — CHANGES_REQUESTED** [review]
Do you mind doc those new routes on OpenAPI file (`docs/reference/api.yaml`)?

**Comment by wpjunior** [comment]
sure @nettoclaudio, I missed this file, I sent changes now

File: api/auth_test.go [hunk 1]
```diff
@@ -1620,3 +1620,76 @@ func (s *AuthSuite) TestUpdateTeamErrorInRollback(c *check.C) {
...
+	c.Assert(recorder.Code, check.Equals, http.StatusOK)
+
+	r := []teamUserItem{}
+	json.NewDecoder(recorder.Body).Decode(&r)
+	c.Assert(r, check.HasLen, 1)
+	c.Assert(r[0].Email, check.Equals, u1.Email)
+	c.Assert(r[0].Roles, check.DeepEquals, []string{role1.Name})
+}
+
+func (s *AuthSuite) TestTeamGroupsList(c *check.C) {
```
Comment (crgisch) [inline api/auth_test.go:1665]: Could you maybe add some tests for other cases that arent the happy path? (Ex: no permission, no team found...)

File: api/server.go [hunk 2]
```diff
@@ -371,6 +371,8 @@ func RunServer(dry bool) http.Handler {
 	m.Add("1.4", http.MethodGet, "/teams/{name}", AuthorizationRequiredHandler(teamInfo))
 	m.Add("1.12", http.MethodGet, "/teams/{name}/quota", AuthorizationRequiredHandler(getTeamQuota))
 	m.Add("1.12", http.MethodPut, "/teams/{name}/quota", AuthorizationRequiredHandler(changeTeamQuota))
+	m.Add("1.16", http.MethodGet, "/teams/{name}/users", AuthorizationRequiredHandler(teamUserList))
```
[the remark this answers is not in the document]
Comment (wpjunior) [reply api/server.go:374]: could be, but this change does not trigger a break version, them we could use like 1.16.4

[the remark this answers is not in the document]
Comment (crgisch) [reply api/server.go:374]: If we are trying to follow semantic versioning, a new feature would be an increase on a minor version, in this case 1.17.0. An increase to PATCH (major.minor.PATCH) usually means fix(es). Dont want to be picky about it, just some food for thought https://semver.org/

[the remark this answers is not in the document]
Comment (wpjunior) [reply api/server.go]: Sure, let's move to 1.17 =)

**Review by crgisch — APPROVED** [review]
LGTM 👍

**Pull request merged**